server.js

```diff
@@ -5425,36 +5425,32 @@ camp.route(/^\/vscode-marketplace\/(d|v|r)\/(.*)\.(svg|png|gif|jpg|json)$/,
 
       try {
         switch (reqType) {
-          case 'd': {
+          case 'd':
             badgeData.text[0] = getLabel('downloads', data);
-            let count = getVscodeStatistic(buffer, 'install');
+            var count = getVscodeStatistic(buffer, 'install');
             badgeData.text[1] = metric(count);
+            badgeData.colorscheme = downloadCountColor(count);
             break;
-          }
-          case 'r': {
+          case 'r':
             badgeData.text[0] = getLabel('rating', data);
-            let rate = getVscodeStatistic(buffer, 'averagerating').toFixed(2);
-            let totalrate = getVscodeStatistic(buffer, 'ratingcount');
+            var rate = getVscodeStatistic(buffer, 'averagerating').toFixed(2);
+            var totalrate = getVscodeStatistic(buffer, 'ratingcount');
             badgeData.text[1] = rate + '/5 (' + totalrate + ')';
+            badgeData.colorscheme = floorCountColor(rate, 2, 3, 4);
             break;
-          }
-          case 'v': {
+          case 'v':
             badgeData.text[0] = getLabel('visual studio marketplace', data);
             var version = buffer.results[0].extensions[0].versions[0].version;
             badgeData.text[1] = versionText(version);
             badgeData.colorscheme = versionColor(version);
             break;
-          }
         }
-
       } catch (e) {
         badgeData.text[1] = 'invalid';
+      } finally {
```

Comment:
We don't usually use finally like this for other badges. I'm not saying we shouldn't but if we refactor the code to have less duplication, it may be trickier to do if we aren't consistent everywhere.